examples/factoranalysis/htfa_cv_example.py

```diff
@@ -0,0 +1,342 @@
+        Data used for testing reconstruction error.
+
+    test_weigth_R : 2D array 
+        Coordinate matrix used for testing weights.
+    
+    test_recon_R : 2D array 
+        Coordinate matrix used for testing reconstruction error.
+
+    centers : 2D array 
+        Center matrix of HTFA factors.
+    
+    widths : 1D array 
+        Width matrix of HTFA factors.
+        
+    Returns
+    -------
+    float 
+        Returns root mean squared error on test.
+
+    """
+
+    # calculate F on test_weight_R, based on trained centers/widths
+    unique_R, inds = htfa.get_unique_R(test_weight_R)
+    F = htfa.get_factors(unique_R,
+                         inds,
+                         centers,
+                         widths)
+    # calculate weights on test_weight_data
+    W = htfa.get_weights(test_weight_data, F)
+
+    # calculate F on final test_recon_data
+    unique_R, inds = htfa.get_unique_R(test_recon_R)
+    F = htfa.get_factors(unique_R,
+                         inds,
+                         centers,
+                         widths)
+    return recon_err(test_recon_data, F, W)
+
+n_subj = 2
+comm = MPI.COMM_WORLD
+rank = comm.Get_rank()
+size = comm.Get_size()
+group_id = int(rank/n_subj)
+n_group = math.ceil(size/n_subj)
+htfa_comm = comm.Split(group_id, rank)
+htfa_rank = htfa_comm.Get_rank()
+htfa_size = htfa_comm.Get_size()
+
+if rank == 0:
+    import logging
+    logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+data_dir = os.path.join(os.getcwd(), 'data')
+if rank == 0 and not os.path.exists(data_dir):
+    os.makedirs(data_dir)
+
+url = []
+url.append(' https://www.dropbox.com/s/r5s9tg4ekxzbrco/s0.mat?dl=0')
+url.append(' https://www.dropbox.com/s/39tr01m76vxwaqa/s1.mat?dl=0')
+
+for idx in range(n_subj):
+    if idx % size == rank:
+        file_name = os.path.join(data_dir, 's' + str(idx) + '.mat')
+        #check if file has already been downloaded
+        if not os.path.exists(file_name):
+            #check if URL exists
+            ret = requests.head(url[idx])
+            if ret.status_code == 200:
+                #download data
+                cmd = 'curl --location -o ' + file_name + url[idx]
+                try:
+                    retcode = call(cmd, shell=True)
+                    if retcode < 0:
+                        print("File download was terminated by signal", -retcode, file=sys.stderr)
+                    else:
+                        print("File download returned", retcode, file=sys.stderr)
+                except OSError as e:
+                    print("File download failed:", e, file=sys.stderr)
+            else:
+                 print("File s%d.mat does not exist!\n"%idx)
+
+#get fMRI data and scanner RAS coordinates
+data = []
+R = []
+mapping = {}
+n_local_subj = 0
+for idx in range(n_subj):
+    if idx % htfa_size == htfa_rank:
+        file_name = os.path.join(data_dir, 's' + str(idx) + '.mat')
+        all_data = scipy.io.loadmat(file_name)
```

Comment:
I tried the example and noticed some file not found errors:
```
$ ./run_mpi_htfa_cv.sh 
About to run command 'mpirun -np 6 python3 htfa_cv_example.py'
Traceback (most recent call last):
  File "htfa_cv_example.py", line 185, in <module>
    all_data = scipy.io.loadmat(file_name)
  File "/home/mihai/.local/lib/python3.5/site-packages/scipy/io/matlab/mio.py", line 135, in loadmat
    MR = mat_reader_factory(file_name, appendmat, **kwargs)
  File "/home/mihai/.local/lib/python3.5/site-packages/scipy/io/matlab/mio.py", line 58, in mat_reader_factory
    byte_stream = _open_file(file_name, appendmat)
  File "/home/mihai/.local/lib/python3.5/site-packages/scipy/io/matlab/mio.py", line 23, in _open_file
    return open(file_like, 'rb')
FileNotFoundError: [Errno 2] No such file or directory: '/home/mihai/Surse/brainiak/examples/factoranalysis/data/s0.mat'
```
I think we need a barrier before this loop, so the threads not involved in the download wait for the download to finish.